odo/backends/avro.py

```diff
@@ -0,0 +1,412 @@
+from __future__ import absolute_import, division, print_function
+
+import errno
+import os
+import uuid
+import json
+import fastavro
+import six
+from avro import schema, datafile, io
+from avro.schema import AvroException
+from multipledispatch import dispatch
+import pandas as pd
+from datashape import dshape, discover, var, Record, Map, Var, \
+    Option, null, string, int8, int32, int64, float64, float32, boolean, bytes_
+import datashape.coretypes as ct
+from collections import Iterator
+from ..append import append
+from ..convert import convert
+from ..resource import resource
+from ..temp import Temp
+
+try:
+    from avro.schema import make_avsc_object as schema_from_dict #Python 2.x
+except ImportError:
+    from avro.schema import SchemaFromJSONData as schema_from_dict  #Python 3.x
+
+PRIMITIVE_TYPES_MAP = {
+    'null': null,
+    'boolean': boolean,
+    'string': string,
+    'bytes': int8,
+    'int': int32,
+    'long': int64,
+    'double': float64,
+    'float': float32,
+}
+
+NAMED_TYPES_MAP = {
+    'fixed': bytes_, #TODO
+    'enum': int8, #TODO
+    'record': Record,
+    'error': Record, #TODO
+}
+
+COMPOUND_TYPES_MAP = {
+    'array': Var,
+    'map': Map,
+    'union': null, #TODO
```

Comment:
What do these `# TODO`s indicate?  Will using `null` here result in valid datashape-to-avro schema translation?